modules/disko/lenovo-x1-disko-luks.nix

```diff
@@ -0,0 +1,9 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{pkgs, ...}: {
+  environment.systemPackages = [pkgs.fido2luks];
+  boot.initrd.luks.fido2Support = true;
+  boot.initrd.luks.devices.encrypted.fido2.credentials = ["b610d7d28dc9f3a87a643e2be6f35df7c7faaa09c2439bbc3679ffd07b726c46898e49f7ac9a1feeed872cea91d830b8"];
```

Comment:
Is this option needed for fido to figure out the right device to use?

Reply:
I understand `credentials` allow multiple [fidos per luks](https://search.nixos.org/options?channel=23.11&show=boot.initrd.luks.devices.%3Cname%3E.fido2.credentials&from=0&size=50&sort=relevance&type=packages&query=fido2).
This raises good questions on:
- if we could include step 1. to the ghaf installer so that some kind of guided wizard would instruct installer to enroll the fido device(s) without manually copying the credential
- how we want to define the installer to guide fido device(s) enrollment? I think it could make sense to guide (or at least provide an option) the user to enroll two fido devices. If one is lost... Like the saying goes - two is one and one is none.

Reply:
> Is this option needed for fido to figure out the right device to use?

Yes as per my understanding credentials option needed to open luks encrypted drive and key should be same what we have used with `fido2luks add-key ...` command. So during NixOS stage 1 there will be decryption/device opening happening [here](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/system/boot/luksroot.nix#L470-L471)

Reply:
> I understand `credentials` allow multiple [fidos per luks](https://search.nixos.org/options?channel=23.11&show=boot.initrd.luks.devices.%3Cname%3E.fido2.credentials&from=0&size=50&sort=relevance&type=packages&query=fido2). This raises good questions on:
> 
> * if we could include step 1. to the ghaf installer so that some kind of guided wizard would instruct installer to enroll the fido device(s) without manually copying the credential
> * how we want to define the installer to guide fido device(s) enrollment? I think it could make sense to guide (or at least provide an option) the user to enroll two fido devices. If one is lost... Like the saying goes - two is one and one is none.

Correct @vilvo if GUI/Wizard installer way will be more user friendly. Or maybe we can have some app in `gui-vm` to view/modify fido2 credentials at later stages of use in terms of adding, deleting as per needs of user.